### PR TITLE
🔨 Fix markdown diagnostics issue with files having dot in their names

### LINKED ISSUE #153094: Experimental markdown validate doesn't work for local links without file extension specified if file name contains period `.` characters
Does this issue occur when all extensions are disabled?: `Yes`

- VS Code Version: (see below)
- OS Version: (see below)

```
Version: 1.68.1 (user setup)
Commit: 30d9c6cd9483b2cc586687151bcbcd635f373630
Date: 2022-06-14T12:48:58.283Z
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Windows_NT x64 10.0.22000
```

Steps to Reproduce:

1. Enable the markdown experimental validation feature: `"markdown.experimental.validate.enabled": true`

1. Create three markdown pages:
    1. `readme.md`
    3. `readme_test.md`
    4. `readme.test.md`

1. In `readme.md` add the following content:

    ```markdown
    - [link to readme_test with working validation](readme_test)
    - [link to readme_test.md with working validation](readme_test.md)
    - [link to readme.test with broken validation](readme.test)
    - [link to readme.test.md with working validation](readme.test.md)
    ```

1. Take note of the validation error as shown below:

    image showing markdown experimental validation error on link containing multiple period characters

### More information

As you can see, this validation error only occurs in one scenario. Although the obvious answer may seem to just specify the file extension, unfortunately this isn't suitable in certain scenarios.

In my case, I am creating relative path links to pages within a Wiki on my GitHub repository. Specifying the file extension results in the `raw` page being displayed, rather than the rendered content.

Obviously, this isn't what we want when creating content, but I don't want to have to suppress these errors.

Your help resolving this issue would be greatly appreciated.

### PR DESCRIPTION
**Pull request by babakks**
This PR fixes #153094

@mjbvz Do you think this simply works? Or there are further complications?

### CODE REVIEW

File: extensions/markdown-language-features/src/languageFeatures/references.ts [hunk 1]
```diff
@@ -341,8 +341,8 @@ export async function tryResolveLinkPath(originalUri: vscode.Uri, workspace: IMd
 		return originalUri;
 	}
 
-	// We don't think the file exists. If it doesn't already have an extension, try tacking on a `.md` and using that instead
-	if (uri.Utils.extname(originalUri) === '') {
+	// We don't think the file exists. If it doesn't already have a `.md` extension, try tacking on a `.md` and using that instead
+	if (uri.Utils.extname(originalUri) !== '.md') {
```
Comment (mjbvz) [inline extensions/markdown-language-features/src/languageFeatures/references.ts:345]: This seems ok. You can use the `looksLikeMarkdownPath` helper for this to handle other md file extensions too: https://github.com/microsoft/vscode/blob/0bc31097613ab4e16c1e4dda85750bc0326e77e6/extensions/markdown-language-features/src/util/file.ts#L25

Maybe we should also skip this check for common image file extensions (`.jpg`, `.png`, `.gif`, ...)

Comment (babakks) [reply extensions/markdown-language-features/src/languageFeatures/references.ts]: Thanks for the hint. I've added a `lookLikeImagePath` helper function to exclude image files too.

**Comment by mjbvz** [comment]
Ah sorry this dropped off my radar

The code for markdown diagnostics code has been moved to this project: https://github.com/microsoft/vscode-markdown-languageservice/

I'm going to close this PR and port over it to that repo

**Comment by babakks** [comment]
@mjbvz Thank you.

**Pull request closed without merging**